rotkehlchen/globaldb/upgrades/v6_v7.py

```diff
@@ -0,0 +1,42 @@
+import logging
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+
+if TYPE_CHECKING:
+    from rotkehlchen.db.drivers.gevent import DBConnection, DBCursor
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+def _create_and_populate_location_asset_mappings_table(cursor: 'DBCursor') -> None:
+    log.debug('Enter _create_and_populate_location_asset_mappings_table')
+
+    cursor.execute("""CREATE TABLE IF NOT EXISTS location_asset_mappings (
+    location TEXT NOT NULL,
+    value TEXT NOT NULL,
+    target_asset TEXT NOT NULL COLLATE NOCASE,
+    FOREIGN KEY(target_asset) REFERENCES assets(identifier) ON UPDATE CASCADE ON DELETE CASCADE,
+    PRIMARY KEY (location, value)
+    );""")
+    cursor.execute(  # populate the mappings for all the exchanges
+        (Path(__file__).resolve().parent.parent.parent / 'data' / 'populate_location_asset_mappings.sql').read_text(encoding='utf8'),  # noqa: E501
```

Comment:
I see a problem with the insert. We add assets in the remote updates but it can happen that the user didn't pull the latest update and then the asset is not in the `assets` table raising a foreign key error in the upgrade. I believe that the target_asset column shouldn't be a foreign key

Reply:
Also can you add some error handling in the execute query? In the event that something really wrong happens

Reply:
the error is handled in `rotkehlchen/globaldb/upgrades/manager.py` like this
```py
        try:
            upgrade.function(connection)
        except BaseException as e:
            # Problem .. restore DB backup, log all info and bail out
            error_message = (
                f'Failed at global DB upgrade from version {upgrade.from_version} to '
                f'{to_version}: {e!s}'
            )
            stacktrace = traceback.format_exc()
            log.error(f'{error_message}\n{stacktrace}')
            shutil.copyfile(tmp_db_path, global_dir / db_filename)
            raise ValueError(error_message) from e
```
should be sufficient right?